### PR TITLE
New REST API resources

### PR DESCRIPTION
**Pull request by Vilz92**
- api/v2/exercises/{exercise_id}/submissions/{submission_id}
- Magical api/v2/me

### CODE REVIEW

**Comment by teemulehtinen** [comment]
In exercise/api/views.py one should use BaseExercise.DoesNotExist (or django.core.exceptions.ObjectDoesNotExist). There seems to also be few cases where tests expect different fields, see http://plustest.cs.hut.fi/jobs/A-plus-test-PR/builds/85/log

**Comment by raphendyr** [comment]
There is mix of commits with message in Finnish and some with English. Also there is set of merging from develop and lot of fixup commits. Could you cleanup the history a bit (rebase -i)?

File: api/urls_v2.py [hunk 1]
```diff
@@ -24,8 +33,8 @@
 
 urlpatterns = [
     url(r'^', include(api.urls, namespace='api')),
-    url(r'^learningobject/$', LearningObjectList.as_view()),
-    url(r'^submission/$', SubmissionList.as_view()),
+
```
Comment (raphendyr) [inline api/urls_v2.py]: I don't see need for new line here.

File: course/api/views.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 from rest_framework import generics, permissions, viewsets
+
```
Comment (raphendyr) [inline course/api/views.py]: I don't see point for newline. All importas are from rest_framework / libraries

File: userprofile/api/serializers.py [hunk 3]
```diff
@@ -26,12 +25,27 @@ class UserSerializer(UserBriefSerialiser):
     Add the details of a user.
     """
 
+    courses = serializers.SerializerMethodField('list_courses')
```
Comment (raphendyr) [inline userprofile/api/serializers.py]: Something like `courses = CourseBriefSerializer(source='enrolled', many=True)` should be enough

File: userprofile/api/serializers.py [hunk 4]
```diff
@@ -1,8 +1,7 @@
 from rest_framework import serializers
-
```
Comment (raphendyr) [inline userprofile/api/serializers.py]: new line here would be nice to keep external libraries and project imports separated.

**Comment by raphendyr** [comment]
I added some comments on the code.

**Pull request merged**